Bio/PDB/PDBList.py

```diff
@@ -271,21 +269,22 @@ def retrieve_pdb_file(
 
         # Get the compressed PDB structure
         pdb_code = pdb_code.lower()
-        archive = {
+        archive_dict = {
             "pdb": f"pdb{pdb_code}.ent.gz",
             "mmCif": f"{pdb_code}.cif.gz",
             "xml": f"{pdb_code}.xml.gz",
             "mmtf": f"{pdb_code}",
             "bundle": f"{pdb_code}-pdb-bundle.tar.gz",
         }
-        archive_fn = archive[file_format]
 
-        if file_format not in archive.keys():
-            raise Exception(
-                f"Specified file_format {file_format} doesn't exists or is not supported. Maybe a "
-                "typo. Please, use one of the following: mmCif, pdb, xml, mmtf, bundle"
+        if file_format not in archive_dict:
+            raise ValueError(
+                f"Specified file_format {file_format} does not exist or is not supported. "
+                f"Please use one of the following: {', '.join(archive_dict.keys())}."
```

Comment:
You can just ``', '.join(archive_dict)`` in place of ``', '.join(archive_dict.keys())``

Sometimes I'd prefer ``', '.join(sorted(archive_dict))`` but that isn't important here given this dictionary is not dynamically defined.

Reply:
I took up your suggestion in the next set of changes. I also added a unit test to check the behavior when an invalid file format is used.